lib/ui/screens/items_list/widgets/items_grid.dart

```diff
@@ -0,0 +1,135 @@
+import 'dart:async';
+
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:pokedex/configs/images.dart';
+import 'package:pokedex/data/repositories/item_repository.dart';
+import 'package:pokedex/domain/entities/item.dart';
+import 'package:pokedex/domain/entities/pokemon.dart';
+import 'package:pokedex/providers/providers.dart';
+import 'package:pokedex/ui/screens/items_list/widgets/item_card.dart';
+import 'package:pokedex/ui/screens/pokedex/widgets/pokemon_card.dart';
+import 'package:pokedex/ui/widgets/main_app_bar.dart';
+import 'package:pokedex/ui/widgets/pokemon_refresh_control.dart';
+import 'package:pokedex/routes.dart';
+
+class ItemsGrid extends StatefulWidget {
+  @override
+  _ItemsGridState createState() => _ItemsGridState();
+}
+
+class _ItemsGridState extends State<ItemsGrid> {
+  static const double _endReachedThreshold = 200;
+
+  final GlobalKey<NestedScrollViewState> _scrollKey = GlobalKey();
+  ScrollController get innerController => _scrollKey.currentState?.innerController;
+
+  @override
+  void initState() {
+    super.initState();
+    scheduleMicrotask(() {
+      context.read(itemStateProvider).getItems(reset: true);
+      innerController?.addListener(_onScroll);
+    });
+  }
+
+  @override
+  void dispose() {
+    innerController?.dispose();
+
+    super.dispose();
+  }
+
+  void _onScroll() {
+    if (innerController != null && !innerController.hasClients) return;
+
+    final thresholdReached = innerController.position.extentAfter < _endReachedThreshold;
+
+    if (thresholdReached) {
+      // Load more!
+      context.read(itemStateProvider).getItems();
```

Comment:
Either add a load more feature or remove it. The list is currently duplicating everything when you scrolled  to the bottom of the list